vnext/template/shared-lib/proj/BuildFlags.props

```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!-- Flags can be added here to affect the compilation of Microsoft.ReactNative -->
+  <PropertyGroup Label="Microsoft.ReactNative Build Flags">
+    <UseWinUI3>{{useWinUI3}}</UseWinUI3>
```

Comment:
Would useNuget be appropriate as well?

Reply:
If usenuget needs build-time behavior, sure. I'm going to keep this PR focused on WinUI though, feel free to add to the template if you have a use for an additional flag :)